src/main/java/com/maarten551/tictactoe_backend/controller/LobbyController.java

```diff
@@ -0,0 +1,147 @@
+package com.maarten551.tictactoe_backend.controller;
+
+import com.maarten551.tictactoe_backend.dto.PlayerExecutionMove;
+import com.maarten551.tictactoe_backend.dto.SelectedLobbyOverview;
+import com.maarten551.tictactoe_backend.exception.*;
+import com.maarten551.tictactoe_backend.logic.GameSessionContainer;
+import com.maarten551.tictactoe_backend.logic.LobbyContainer;
+import com.maarten551.tictactoe_backend.logic.PlayerContainer;
+import com.maarten551.tictactoe_backend.model.Lobby;
+import com.maarten551.tictactoe_backend.model.Player;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.handler.annotation.*;
+import org.springframework.messaging.simp.SimpMessageHeaderAccessor;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.messaging.simp.annotation.SendToUser;
+import org.springframework.messaging.simp.annotation.SubscribeMapping;
+import org.springframework.stereotype.Controller;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Controller
+public class LobbyController {
+	public static final String LOBBY_OVERVIEW_ENDPOINT = "/lobbies";
+    public static final String LOBBY_JOINED_ENDPOINT = "/lobby/{lobbyId}";
+
+    private final LobbyContainer lobbyContainer;
+    private final PlayerContainer playerContainer;
+    private final GameSessionContainer gameSessionContainer;
+    private final SimpMessagingTemplate template;
+
+	@Autowired
+    public LobbyController(LobbyContainer lobbyContainer, PlayerContainer playerContainer, GameSessionContainer gameSessionContainer, SimpMessagingTemplate template) {
+		this.lobbyContainer = lobbyContainer;
+		this.playerContainer = playerContainer;
+        this.gameSessionContainer = gameSessionContainer;
+        this.template = template;
+	}
+
+	@SubscribeMapping(LOBBY_OVERVIEW_ENDPOINT)
+	public List<Lobby> onLobbyOverview() {
+		return this.lobbyContainer.getAllWaitingLobbies();
+	}
+
+//	@SubscribeMapping(LOBBY_JOINED_ENDPOINT)
+//	public Lobby onJoinLobby(@DestinationVariable String lobbyId) throws LobbyDoesNotExistException {
+//		return this.lobbyContainer.getLobbyById(UUID.fromString(lobbyId));
+//	}
+
+	@MessageMapping({ "/send/lobbies/create" })
+	@SendTo(LOBBY_OVERVIEW_ENDPOINT)
+	public List<Lobby> onLobbyCreate(@Payload String lobbyName, SimpMessageHeaderAccessor headerAccessor)
```

Comment:
I am not familiar with "on" in function names.

Reply:
Is this function used anywhere? Also, why does it return a list?

Reply:
> I am not familiar with "on" in function names.

I just copied this naming behavior from a tutorial, I believe I'm not consistent with this behavior.


> Is this function used anywhere? Also, why does it return a list?

When a new lobby is created, a new message is send to channel inside the constant `LOBBY_OVERVIEW_ENDPOINT`, this message consists of all the waiting lobbies in JSON format. The Spring framework does a lot of translating for you.